metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -2908,7 +2928,14 @@ def emit_dashboards(self) -> Iterable[MetadataWorkUnit]:
             c.DASHBOARDS_CONNECTION,
             dashboards_filter,
         ):
-            yield from self.emit_dashboard(dashboard, dashboard.get(c.WORKBOOK))
+            # LUID is blank if the dashboard is hidden in the workbook.
+            # More info here: https://help.tableau.com/current/api/metadata_api/en-us/reference/dashboard.doc.html
+            if self.config.ingest_hidden_assets or dashboard.get(c.LUID):
+                yield from self.emit_dashboard(dashboard, dashboard.get(c.WORKBOOK))
+            else:
+                logger.debug(
+                    f"Skip dashboard {dashboard.get(c.ID)} because it's hidden (luid is blank)."
+                )
 
     def get_tags(self, obj: dict) -> Optional[List[str]]:
```

Comment:
Why not changing the signature with `get_tags(self, obj: dict) -> Optional[List[str]]` and `return []` at the end.
It will be more coherent with actual calls now.